Bugzilla/App/OAuth2/Provider/Clients.pm

```diff
@@ -68,19 +68,29 @@ sub create {
   my $description = $self->param('description');
   my $id          = $self->param('id');
   my $secret      = $self->param('secret');
+  my $hostname    = $self->param('hostname');
   my $scopes      = $self->every_param('scopes');
   $description
-    or return $self->code_error('param_required', {param => 'description'});
-  $id     or return $self->code_error('param_required', {param => 'id'});
-  $secret or return $self->code_error('param_required', {param => 'secret'});
+    or return $self->code_error('param_required',
+    {function => 'admin/oauth/provider/create', param => 'description'});
+  $id
+    or return $self->code_error('param_required',
+    {function => 'admin/oauth/provider/create', param => 'id'});
+  $secret
+    or return $self->code_error('param_required',
+    {function => 'admin/oauth/provider/create', param => 'secret'});
+  $hostname
+    or return $self->code_error('param_required',
+    {function => 'admin/oauth/provider/create', param => 'hostname'});
```

Comment:
What does the extra `function` parameter do here?

Reply:
It allows the error message to be show a more exact location in which the error occurred. So in this example, the user will know that the endpoint was missing the required parameter and value.

The error message `param_required` is a small portion of a larger template at `templates/en/default/global/code-error.html.tmpl` and is used in many places. So the function variable in the template could also display a method or code function instead of an endpoint. For example `Bugzilla::Bug->create()', etc.